### PR TITLE
RIA-3690 Remission decision notifications for partially-approved and rejected

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3690](https://tools.hmcts.net/jira/browse/RIA-3690)

### Change description ###
Remission notification for partially-approved and rejected

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-3690-remission-decision-partially-approved.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        "reference": "3690_REMISSION_DECISION_PARTIALLY_APPROVED_LEGAL_REPRESENTATIVE",
+        "recipient": "{$TEST_LAW_FIRM_A_USERNAME}",
+        "subject": "Immigration and Asylum appeal: Remission application partially approved",
+        "body": [
+          "PA/12345/2019",
+          "CASE001",
+          "Talha Awan",
+          "{$iaExUiFrontendUrl}",
+          "40.00"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3690-remission-decision-partially-approved.json:49]: The email template in GovNotify is incorrect for Legal Representative. You need to have placeholders in the template for the customer services telephone and email address. You will check the values in the functional test here like this:

`"{$customerServices.telephoneNumber}",
  "{$customerServices.emailAddress}"`

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -570,6 +570,15 @@ govnotify:
     requestNewHearingRequirements:
       legalRep:
         email: fd709bf4-095c-4734-8f88-9a9feba9fd81
+    recordRemissionDecision:
+      partiallyApproved:
+        legalRep:
+          email: 717b0290-36eb-4adf-a4c5-baacf59d379d
```
Comment (bradsorour) [inline src/main/resources/application.yaml]: This template is incorrect in GovNotify. You need to add placeholders for customer services telephone number and email address.

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -570,6 +570,15 @@ govnotify:
...
+        adminOfficer:
+          email: 954c2959-e13a-4907-be1e-3eb9831fc68c
+      rejected:
+        legalRep:
+          email: 8a48b422-c6ed-448d-aec0-a5b43bd2b472
```
Comment (bradsorour) [inline src/main/resources/application.yaml]: Similar to above, you need to add placeholders to this template for customer services telephone number and email address.

File: src/functionalTest/resources/scenarios/RIA-3690-remission-decision-rejected.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      {
+        "reference": "3690_REMISSION_DECISION_REJECTED_LEGAL_REPRESENTATIVE",
+        "recipient": "{$TEST_LAW_FIRM_A_USERNAME}",
+        "subject": "\tImmigration and Asylum appeal: Remission application rejected",
+        "body": [
+          "PA/12345/2019",
+          "CASE001",
+          "Talha Awan",
+          "{$iaExUiFrontendUrl}"
+        ]
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3690-remission-decision-rejected.json:42]: Similar to above, once the GovNotify template has been updated, you will need to check for the following in the functional test:

"{$customerServices.telephoneNumber}",
"{$customerServices.emailAddress}"

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+@Component
+public class LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation implements EmailNotificationPersonalisation {
+
+    private final String partiallyApprovedTemplateId;
+    private final String iaExUiFrontendUrl;
+
+    public LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation(
+        @Value("${govnotify.template.recordRemissionDecision.partiallyApproved.legalRep.email}") String partiallyApprovedTemplateId,
+        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl
+
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java:25]: You need to inject the `CustomerServicesProvider customerServicesProvider` in order to have access to the customer services telephone and email address.

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        return caseId + "_REMISSION_DECISION_PARTIALLY_APPROVED_LEGAL_REPRESENTATIVE";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        return ImmutableMap
+            .<String, String>builder()
+            .put("appealReferenceNumber", asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java:54]: Similar to the other personalisation classes for legal rep, you will need to add the following to the map `.putAll(customerServicesProvider.getCustomerServicesPersonalisation())`

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisation.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+@Component
+public class LegalRepresentativeRemissionDecisionRejectedPersonalisation implements EmailNotificationPersonalisation {
+
+    private final String rejectedTemplateId;
+    private final String iaExUiFrontendUrl;
+
+    public LegalRepresentativeRemissionDecisionRejectedPersonalisation(
+        @Value("${govnotify.template.recordRemissionDecision.rejected.legalRep.email}") String rejectedTemplateId,
+        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl
+
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisation.java:24]: Same comments as above apply to this class.

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisationTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        Map<String, String> personalisation =
+            legalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.getPersonalisation(asylumCase);
+
+        assertEquals(appealReferenceNumber, personalisation.get("appealReferenceNumber"));
+        assertEquals(legalRepRefNumber, personalisation.get("legalRepReferenceNumber"));
+        assertEquals(appellantGivenNames, personalisation.get("appellantGivenNames"));
+        assertEquals(appellantFamilyName, personalisation.get("appellantFamilyName"));
+        assertEquals(iaExUiFrontendUrl, personalisation.get("linkToOnlineService"));
+        assertEquals(amountLeftToPay, personalisation.get("feeAmount"));
+    }
```
Comment (bradsorour) [inline src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisationTest.java:73]: You will need to test for customer services telephone and email address.

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisationTest.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+        Map<String, String> personalisation =
+            legalRepresentativeRemissionDecisionRejectedPersonalisation.getPersonalisation(asylumCase);
+
+        assertEquals(appealReferenceNumber, personalisation.get("appealReferenceNumber"));
+        assertEquals(legalRepRefNumber, personalisation.get("legalRepReferenceNumber"));
+        assertEquals(appellantGivenNames, personalisation.get("appellantGivenNames"));
+        assertEquals(appellantFamilyName, personalisation.get("appellantFamilyName"));
+        assertEquals(iaExUiFrontendUrl, personalisation.get("linkToOnlineService"));
+    }
```
Comment (bradsorour) [inline src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisationTest.java:70]: Similar to above, you will need to test for customer services telephone and email address.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (divisathyan) [reply src/main/resources/application.yaml]: Updated template in Gov notify

File: src/main/resources/application.yaml (hunk 3 above)
Comment (divisathyan) [reply src/main/resources/application.yaml]: Updated template in Gov notify

File: src/functionalTest/resources/scenarios/RIA-3690-remission-decision-partially-approved.json (hunk 1 above)
Comment (divisathyan) [reply src/functionalTest/resources/scenarios/RIA-3690-remission-decision-partially-approved.json:51]: Added the telephoneNumber and emailAddress. Modified the template in Gov Notify

File: src/functionalTest/resources/scenarios/RIA-3690-remission-decision-rejected.json (hunk 4 above)
Comment (divisathyan) [reply src/functionalTest/resources/scenarios/RIA-3690-remission-decision-rejected.json:44]: Added the telephoneNumber and emailAddress. Modified the template in Gov Notify

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java (hunk 5 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java:34]: Changed

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java (hunk 6 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisation.java:65]: Added the telephoneNumber and emailAddress. Modified the template in Gov Notify

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisation.java (hunk 7 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisation.java:27]: same fix

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisationTest.java (hunk 8 above)
Comment (divisathyan) [reply src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionPartiallyApprovedPersonalisationTest.java:92]: Added the telephoneNumber and emailAddress. Modified the template in Gov Notify

File: src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisationTest.java (hunk 9 above)
Comment (divisathyan) [reply src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/legalrepresentative/LegalRepresentativeRemissionDecisionRejectedPersonalisationTest.java:86]: Added the telephoneNumber and emailAddress. Modified the template in Gov Notify

**Review by bradsorour — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**